airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -136,35 +136,25 @@ def read(
                     )
                     logger.info(f"Marking stream {configured_stream.stream.name} as STOPPED")
                     yield stream_status_as_airbyte_message(configured_stream.stream, AirbyteStreamStatus.COMPLETE)
-                except AirbyteTracedException as e:
-                    yield from self._emit_queued_messages()
-                    logger.exception(f"Encountered an exception while reading stream {configured_stream.stream.name}")
-                    logger.info(f"Marking stream {configured_stream.stream.name} as STOPPED")
-                    yield stream_status_as_airbyte_message(configured_stream.stream, AirbyteStreamStatus.INCOMPLETE)
-                    yield e.as_sanitized_airbyte_message(stream_descriptor=StreamDescriptor(name=configured_stream.stream.name))
-                    stream_name_to_exception[stream_instance.name] = e
-                    if self.stop_sync_on_stream_failure:
-                        logger.info(
-                            f"Stopping sync on error from stream {configured_stream.stream.name} because {self.name} does not support continuing syncs on error."
-                        )
-                        break
                 except Exception as e:
                     yield from self._emit_queued_messages()
                     logger.exception(f"Encountered an exception while reading stream {configured_stream.stream.name}")
                     logger.info(f"Marking stream {configured_stream.stream.name} as STOPPED")
                     yield stream_status_as_airbyte_message(configured_stream.stream, AirbyteStreamStatus.INCOMPLETE)
-                    display_message = stream_instance.get_error_display_message(e)
-                    stream_descriptor = StreamDescriptor(name=configured_stream.stream.name)
-                    if display_message:
-                        traced_exception = AirbyteTracedException.from_exception(
-                            e, message=display_message, stream_descriptor=stream_descriptor
-                        )
+
+                    if isinstance(e, AirbyteTracedException):
+                        stream_descriptor = StreamDescriptor(name=configured_stream.stream.name)
+                        traced_exception = e
+                        info_message = f"Stopping sync on error from stream {configured_stream.stream.name} because {self.name} does not support continuing syncs on error."
                     else:
-                        traced_exception = AirbyteTracedException.from_exception(e, stream_descriptor=stream_descriptor)
-                    yield traced_exception.as_sanitized_airbyte_message()
+                        stream_descriptor = None
```

Comment:
Why don't we populate the stream_descriptor if the exception is not a AirbyteTracedException? It feels like it would still be useful when calling `traced_exception.as_sanitized_airbyte_message(stream_descriptor=stream_descriptor)`

Reply:
I just follow to the original solution - https://github.com/airbytehq/airbyte/blame/efd76b79d05bf96e5e66c72169c954e4639b4660/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L167C3-L167C3

But I agree with your solution

Reply:
The stream descriptor was added as part of `from_exception` in the previous solution so I think it was available